Switch formatter to ruff because isort is having stability issues.

ctheune opened the pull request:
isort due to a bug (https://github.com/PyCQA/isort/issues/1889) is giving us different results between CLI/Github Actions and the pre-commit hook.

Switching all linting and formatting to ruff gets us back to a stable situation.

@flyingcircusio/release-managers

## Release process

- [ ] Created changelog entry using `./changelog.sh`

no, internal reformatting only

## PR release workflow (internal)

- [ ] PR has internal ticket
- [ ] internal issue ID (PL-…) part of branch name
- [ ] internal issue ID mentioned in PR description text
- [ ] ticket is on Platform agile board
- [ ] ticket state set to *Pull request ready*
- [ ] if ticket is more urgent than within the next few days, directly contact a member of the Platform team
<!---->
- [ ] set urgency and risk labels
- [ ] ensure the merge bot has determined a merge date
- [ ] ensure all checks are green
- [ ] get a review from a colleague

skipping this, as we need this to unblock actual changes


## Design notes

- [ ] Provide a feature toggle if the change might need to be adjusted/reverted quickly depending on context. Consider whether the default should be `on` or `off`. Example: rate limiting.
- [ ] All customer-facing features and (NixOS) options need to be discoverable from documentation. Add or update relevant documentation such that hosted and guided customers can understand it as well.

not applicable

## Security implications

- [x] [Security requirements](https://wiki.flyingcircus.io/System_Development_Guideline#Security_requirement_principles_and_testing) defined? (WHERE)
- [ ] Security requirements tested? (EVIDENCE)

only reformatting

ctheune committed:
Switch formatter to ruff because isort is having stability issues.

isort due to a bug (https://github.com/PyCQA/isort/issues/1889)
is giving us different results between CLI/Github Actions and the
pre-commit hook.

Switching all linting and formatting to ruff gets us back to a stable
situation.

osnyx commented on pkgs/fc/ceph/src/fc/ceph/backup.py:
```diff
@@ -46,9 +45,10 @@ def create(
             "-f",
-            "-L", self.name,
+            "-L",
+            self.name,
```
This is missing the fmt: off -> fmt: skip adjusting

osnyx reviewed (commented):
Found one missing fmt: skip adjustment.

I'm going to fix it myself and, as I am going to touch this anyways, make use of `.git-blame-ignore-revs`.

osnyx committed:
treewide: apply new ruff formatting

includes some manual adjustments of "fmt: skip" and "fmt: on/off"
directives to account for slight differences between black and ruff.

osnyx committed:
git-blame-ignore-revs: add ruff reformatting
